src/Command/Push/PushCodeCommand.php

```diff
@@ -34,10 +41,160 @@ protected function configure() {
+      ->ignoreVCSIgnored(TRUE);
+    $targetFinder = Finder::create();
+    $targetFinder->files()->in($artifact_dir)->ignoreDotFiles(FALSE);
+    $this->localMachineHelper->getFilesystem()->mirror($this->dir, $artifact_dir, $originFinder, ['override' => TRUE, 'delete' => TRUE], $targetFinder);
+
+    $this->logger->info('Installing Composer production dependencies');
+    $this->localMachineHelper->executeFromCmd('composer install --no-dev --no-interaction --optimize-autoloader', $output_callback, $artifact_dir, $this->output->isVerbose());
+  }
+
+  protected function sanitize(string $artifact_dir):void {
+    $this->logger->info('Finding Drupal core text files');
+    $sanitizeFinder = Finder::create()
+      ->files()
+      ->name('*.txt')
+      ->notName('LICENSE.txt')
+      ->in("$artifact_dir/docroot/core");
+
+    $this->logger->info('Finding VCS directories');
+    $vcsFinder = Finder::create()
+      ->ignoreDotFiles(FALSE)
+      ->ignoreVCS(FALSE)
+      ->directories()
+      ->in(["$artifact_dir/docroot",
+        "$artifact_dir/vendor",
+      ])
+      ->name('.git');
+    $drush_dir = "$artifact_dir/drush";
+    if (file_exists($drush_dir)) {
+      $vcsFinder->in($drush_dir);
+    }
+    if ($vcsFinder->hasResults()) {
+      $sanitizeFinder->append($vcsFinder);
+    }
+
+    $this->logger->info('Finding INSTALL database text files');
+    $dbInstallFinder = Finder::create()
+      ->files()
+      ->in([$artifact_dir])
+      ->name('/INSTALL\.[a-z]+\.(md|txt)$/');
+    if ($dbInstallFinder->hasResults()) {
+      $sanitizeFinder->append($dbInstallFinder);
+    }
+
+    $this->logger->info('Finding other common text files');
+    $filenames = [
+      'AUTHORS',
+      'CHANGELOG',
+      'CONDUCT',
+      'CONTRIBUTING',
+      'INSTALL',
+      'MAINTAINERS',
+      'PATCHES',
+      'TESTING',
+      'UPDATE',
+    ];
+    $textFileFinder = Finder::create()
+      ->files()
+      ->in(["$artifact_dir/docroot"])
+      ->name('/(' . implode('|', $filenames) . ')\.(md|txt)$/');
+    if ($textFileFinder->hasResults()) {
+      $sanitizeFinder->append($textFileFinder);
+    }
+
+    $this->logger->info("Removing sanitized files from build");
+    $this->localMachineHelper->getFilesystem()->remove($sanitizeFinder);
+  }
+
+  protected function push(Closure $output_callback, string $artifact_dir, string $commit_hash):void {
+    $this->logger->info('Adding and committing changed files');
+    $force_add = [
+      'vendor',
+      'docroot/core',
+      'docroot/modules/contrib',
+      'docroot/themes/contrib',
+      'docroot/profiles/contrib',
+      'docroot/libraries'
+    ];
+    $this->localMachineHelper->executeFromCmd('git add -A', $output_callback, $artifact_dir, $this->output->isVerbose());
+    $this->localMachineHelper->executeFromCmd('git add -f ' . implode(' ', $force_add), $output_callback, $artifact_dir, $this->output->isVerbose());
```

Comment:
I think it'd be good to emit a message showing exactly what we're force committing.